### ISSUE TITLE
Add Gradient Clipping

### ISSUE THREAD

**Issue by tokenbender** [tokenbender/avataRL#27]
## Description
Add gradient clipping to prevent training instability from gradient explosion.

## Tasks
- [ ] Add GRAD_CLIP parameter to config.py (default: 1.0)
- [ ] Implement gradient clipping in train_step after unscaling
- [ ] Make clipping optional (GRAD_CLIP = 0.0 disables)
- [ ] Test with GRPO algorithm to ensure compatibility

## Priority
High - Prevents training instability

## Technical Details
- Add after scaler.unscale_(optimizer):
```python
if GRAD_CLIP != 0.0:
    torch.nn.utils.clip_grad_norm_(model.parameters(), GRAD_CLIP)
```
- Consider per-stage gradient clipping thresholds
- Monitor gradient norms in logging

**Comment by tokenbender** [comment]
Completed: Added gradient clipping support to train_core.py

Changes made:
- Added GRAD_CLIP = 1.0 configuration parameter in config.py
- Made gradient clipping conditional in train_core.py (only applies when GRAD_CLIP != 0.0)
- Implementation matches train_modal_standalone.py pattern with proper mixed precision support

The gradient clipping is now configurable and can be disabled by setting GRAD_CLIP = 0.0 in the config.
